### PR TITLE
fix(orchestrator): repair sql injection

### PR DESCRIPTION
**Pull request by jing-zhi**
#### What this PR does / why we need it:
Bugfix： repair sql injection （修复了orchestrator instances-usage sql 注入问题）

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=507824&iterationID=-1&type=TASK)

#### Specified Reviewers:

/assign @chengjoey 

#### ChangeLog

Bugfix： repair sql injection （修复了orchestrator instances-usage sql 注入问题）

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     repair sql injection - application_id         |
| 🇨🇳 中文    |       修复sql注入-application_id 问题       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

/cherry-pick release/2.4

### CODE REVIEW

File: internal/tools/orchestrator/scheduler/instanceinfo/instance.go [hunk 1]
```diff
@@ -77,7 +78,8 @@ func (r *InstanceReader) ByEdgeSite(name string) *InstanceReader {
 	return r
 }
 func (r *InstanceReader) ByApplicationID(id string) *InstanceReader {
-	r.conditions = append(r.conditions, fmt.Sprintf("application_id = \"%s\"", id))
+	r.conditions = append(r.conditions, "application_id = ?")
+	r.values = append(r.values, id)
 	return r
 }
```
Comment (chengjoey) [inline internal/tools/orchestrator/scheduler/instanceinfo/instance.go:84]: how about `ByNamespace`, `ByNamespace` and other conditions

File: internal/tools/orchestrator/scheduler/instanceinfo/instance.go [hunk 2]
```diff
@@ -161,7 +164,7 @@ func (r *InstanceReader) Limit(n int) *InstanceReader {
...
 func (r *InstanceReader) Do() ([]InstanceInfo, error) {
 	instanceinfo := []InstanceInfo{}
-	expr := r.db.Where(strutil.Join(r.conditions, " AND ", true)).Order("started_at desc")
+	expr := r.db.Where(strutil.Join(r.conditions, " AND ", true), r.values[0], r.values[1]).Order("started_at desc")
```
Comment (chengjoey) [inline internal/tools/orchestrator/scheduler/instanceinfo/instance.go:167]: other conditions `runtime` `service` may be not have values, so values[0], values[idx] and etc will cause panic

**Review by chengjoey — APPROVED** [review]

File: internal/tools/orchestrator/scheduler/instanceinfo/instance.go (hunk 1 above)
Comment (sfwn) [reply internal/tools/orchestrator/scheduler/instanceinfo/instance.go:85]: All `ByXXX` method should be fixed.

Comment (sfwn) [reply internal/tools/orchestrator/scheduler/instanceinfo/instance.go:85]: All `ByXXX` method should be fixed.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**